openmetadata-ui/src/main/resources/ui/src/context/PermissionProvider/PermissionProvider.tsx

```diff
@@ -180,7 +180,7 @@ const PermissionProvider: FC<PermissionProviderProps> = ({ children }) => {
     if (isEmpty(currentUser)) {
       resetPermissions();
     }
-  }, [currentUser]);
+  }, [currentUser?.teams, currentUser?.roles]);
 
```

Comment:
wouldn't this change affect globally? have you checked the behaviour change due to this on other places?

Reply:
yes, this won't effect other places since this will only be required for updating the user `Role` from the `Team and roles added or removed`